api/src/pcapi/core/users/models.py

```diff
@@ -339,33 +338,33 @@ def setPassword(self, newpass):  # type: ignore [no-untyped-def]
         self.password = crypto.hash_password(newpass)
 
     @property
-    def age(self) -> Optional[int]:
+    def age(self) -> int | None:
         return users_utils.get_age_from_birth_date(self.dateOfBirth.date()) if self.dateOfBirth is not None else None
 
     @property
-    def deposit(self) -> Optional["Deposit"]:
+    def deposit(self) -> typing.Optional["Deposit"]:
```

Comment:
(question) Est-ce voulu d'avoir laissé `Optional` ici ? J'ai l'impression que le typage suivant convient à mypy : 

```suggestion
    def deposit(self) -> "Deposit" | None:
```

Idem à 2-3 autres endroits trouvé en greppant `Optional` dans ta branche.

Reply:
malheureusement `"type" | None` ne marche pas. `"type  | None"`  semble marcher mais il faut que je fasse plus de tests pour le valider

Reply:
c'est pas vraiment voulu de les avoir laissé, plus que je n'ai pas réussis a les enlever